packages/blockchain-wallet-v4-frontend/src/data/analytics/types/swap.ts

```diff
@@ -0,0 +1,22 @@
+export enum Events {
+  SWAP_EARN_REWARDS_BUTTON_CLICKED = 'Swap Earn Rewards Clicked',
+  SWAP_EARN_REWARDS_BUTTON_VIEWED = 'Swap Earn Rewards Viewed'
+}
+
+type SwapEarnRewardsButtonClick = {
+  key: Events.SWAP_EARN_REWARDS_BUTTON_CLICKED
+  properties: {
+    currency: string
+    device: string
+  }
+}
+
+type SwapEarnRewardsButtonView = {
+  key: Events.SWAP_EARN_REWARDS_BUTTON_VIEWED
+  properties: {
+    currency: string
+    device: string
+  }
+}
+
+export type TrackEventAction = SwapEarnRewardsButtonClick | SwapEarnRewardsButtonView
```

Comment:
Events added to the library
![image](https://user-images.githubusercontent.com/97299628/171927516-0045b1d9-7527-4b3d-a6ec-0340f6df4c03.png)